### PR TITLE
php@8.1 8.1.16

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/php@8.1.rb [hunk 1]
```diff
@@ -63,7 +63,6 @@ class PhpAT81 < Formula
 
   on_macos do
     # PHP build system incorrectly links system libraries
-    # see https://github.com/php/php-src/pull/3472
```
Comment (SMillerDev) [inline Formula/php@8.1.rb:66]: No, this should be fixed instead

**Comment by p-linnane** [comment]
Audit failure due to https://github.com/Homebrew/brew/pull/14326.

**Comment by carlocab** [comment]
Let's see what happens when we drop the patch. Here's the current linkage on Big Sur/Monterey:
```
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libbz2.1.0.dylib
  /usr/lib/libc++.1.dylib
  /usr/lib/libedit.3.dylib
  /usr/lib/libexpat.1.dylib
  /usr/lib/libexslt.0.dylib
  /usr/lib/libffi.dylib
  /usr/lib/libiconv.2.dylib
  /usr/lib/libicucore.A.dylib
  /usr/lib/libncurses.5.4.dylib
  /usr/lib/libnetwork.dylib
  /usr/lib/libresolv.9.dylib
  /usr/lib/libxml2.2.dylib
  /usr/lib/libxslt.1.dylib
  /usr/lib/libz.1.dylib
```
Ventura is the same except it lacks `/usr/lib/libicucore.A.dylib` (no idea why).

**Comment by carlocab** [comment]
Well, that broke quickly...
```
  Undefined symbols for architecture arm64:
    "_ldap_create_passwordpolicy_control", referenced from:
        _zif_ldap_exop_passwd in ldap.o
    "_ldap_create_sort_control_value", referenced from:
        __php_ldap_controls_from_array in ldap.o
  Undefined symbols for architecture arm64:
    "_ldap_create_passwordpolicy_control", referenced from:
        _zif_ldap_exop_passwd in ldap.o
    "_ldap_create_vlv_control_value", referenced from:
        __php_ldap_controls_from_array in ldap.o
    "_ldap_create_sort_control_value", referenced from:
        __php_ldap_controls_from_array in ldap.o
    "_ldap_destroy", referenced from:
        _ldap_link_free_obj in ldap.o
        _zif_ldap_unbind in ldap.o
    "_ldap_create_vlv_control_value", referenced from:
        __php_ldap_controls_from_array in ldap.o
    "_ldap_destroy", referenced from:
        _ldap_link_free_obj in ldap.o
        _zif_ldap_unbind in ldap.o
    "_ldap_parse_passwordpolicy_control", referenced from:
        __php_ldap_controls_to_array in ldap.o
  Undefined symbols for architecture arm64:
    "_ldap_create_passwordpolicy_control", referenced from:
        _zif_ldap_exop_passwd in ldap.o
    "_ldap_parse_passwordpolicy_control", referenced from:
        __php_ldap_controls_to_array in ldap.o
    "_tidyLibraryVersion", referenced from:
        _zm_info_tidy in tidy.o
    "_ldap_create_sort_control_value", referenced from:
        __php_ldap_controls_from_array in ldap.o
    "_tidyLibraryVersion", referenced from:
        _zm_info_tidy in tidy.o
    "_ldap_create_vlv_control_value", referenced from:
        __php_ldap_controls_from_array in ldap.o
  ld: symbol(s) not found for architecture arm64
  ld: symbol(s) not found for architecture arm64
    "_ldap_destroy", referenced from:
        _ldap_link_free_obj in ldap.o
        _zif_ldap_unbind in ldap.o
  
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by p-linnane** [comment]
Given that the patch is still necessary despite the PR being closed, how do we want to handle this? Others have been removing `https://github.com` from these links to circumvent the audit, but that kind of defeats the purpose. See also: https://github.com/Homebrew/homebrew-core/pull/123252

**Comment by SMillerDev** [comment]
I'd say we should open an issue upstream to try and fix this again.

**Comment by No0ne** [comment]
There seems to be a file checksum mismatch 🤔
```
noone@macbookair ~ % curl -s https://www.php.net/distributions/php-8.1.16.tar.xz | shasum -a 256
d61f13d96a58b93c39672b58f25e1ee4ce88500f4acb1430cb01a514875c1258  -
noone@macbookair ~ % curl -s https://fossies.org/linux/www/php-8.1.16.tar.xz | shasum -a 256
7108b7347981ad6e610aaf3b3fb0f6444019ab6f59a872c1b55a29bc753eba93  -
noone@macbookair ~ % 
```
The difference between the tar files is this:
```
noone@macbookair Desktop % diff -ru aa bb
Only in bb/ext/bz2/tests: 003私はガラスを食べられます.txt.bz2
Only in aa/ext/bz2/tests: 003私はガラスを食べられます.txt.bz2
Only in bb/ext/exif/tests: bug34704私はガラスを食べられます.jpg
Only in aa/ext/exif/tests: bug34704私はガラスを食べられます.jpg
Only in bb/ext/exif/tests: bug68113私はガラスを食べられます.jpg
Only in aa/ext/exif/tests: bug68113私はガラスを食べられます.jpg
Only in bb/ext/exif/tests: test2私はガラスを食べられます.jpg
Only in aa/ext/exif/tests: test2私はガラスを食べられます.jpg
Only in bb/ext/fileinfo/tests: bug71527私はガラスを食べられます.magic
Only in aa/ext/fileinfo/tests: bug71527私はガラスを食べられます.magic
Only in bb/ext/fileinfo/tests: cve-2014-1943私はガラスを食べられます.data
Only in bb/ext/fileinfo/tests: cve-2014-1943私はガラスを食べられます.magic
Only in aa/ext/fileinfo/tests: cve-2014-1943私はガラスを食べられます.data
Only in aa/ext/fileinfo/tests: cve-2014-1943私はガラスを食べられます.magic
Only in bb/ext/fileinfo/tests: cve-2014-3538私はガラスを食べられます.data
Only in aa/ext/fileinfo/tests: cve-2014-3538私はガラスを食べられます.data
Only in bb/ext/fileinfo/tests: magic私はガラスを食べられます
Only in aa/ext/fileinfo/tests: magic私はガラスを食べられます
Only in bb/ext/gd/tests: Tuffy私はガラスを食べられます.ttf
Only in aa/ext/gd/tests: Tuffy私はガラスを食べられます.ttf
Only in bb/ext/gd/tests: bug22544私はガラスを食べられます.png
Only in aa/ext/gd/tests: bug22544私はガラスを食べられます.png
Only in bb/ext/gd/tests: bug37346私はガラスを食べられます.gif
Only in aa/ext/gd/tests: bug37346私はガラスを食べられます.gif
Only in bb/ext/gd/tests: conv_test私はガラスを食べられます.jpg
Only in aa/ext/gd/tests: conv_test私はガラスを食べられます.jpg
Only in bb/ext/gd/tests: invalid_neg_size私はガラスを食べられます.gd2
Only in aa/ext/gd/tests: invalid_neg_size私はガラスを食べられます.gd2
Only in bb/ext/gd/tests: libgd00094私はガラスを食べられます.xbm
Only in aa/ext/gd/tests: libgd00094私はガラスを食べられます.xbm
Only in bb/ext/gd/tests: src私はガラスを食べられます.wbmp
Only in aa/ext/gd/tests: src私はガラスを食べられます.wbmp
Only in bb/ext/libxml/tests: bug69753私はガラスを食べられます.xml
Only in aa/ext/libxml/tests: bug69753私はガラスを食べられます.xml
Only in bb/ext/simplexml/tests: sxe私はガラスを食べられます.xml
Only in aa/ext/simplexml/tests: sxe私はガラスを食べられます.xml
Only in bb/ext/tidy/tests: 005私はガラスを食べられます.html
Only in aa/ext/tidy/tests: 005私はガラスを食べられます.html
Only in bb/ext/xsl/tests: 私はガラスを食べられますstreamsinclude.xsl
Only in aa/ext/xsl/tests: 私はガラスを食べられますstreamsinclude.xsl
Only in bb/ext/zip/tests: bug40228私はガラスを食べられます.zip
Only in aa/ext/zip/tests: bug40228私はガラスを食べられます.zip
Only in bb/ext/zlib/tests: 004私はガラスを食べられます.txt.gz
Only in aa/ext/zlib/tests: 004私はガラスを食べられます.txt.gz
noone@macbookair Desktop %
```

**Comment by derrabus** [comment]
Why was this PR closed? PHP 8.1.16 was a security release. I understand that the local patch should be taken upstream, but this should not block bumping this formula, should it?

**Comment by chenrui333** [comment]
Alright, as the original branch was deleted, gonna carry the torch in [this new PR](https://github.com/Homebrew/homebrew-core/pull/124040)

**Pull request closed without merging**